### DISCUSSION TITLE
help with search_data() method

### DISCUSSION THREAD

**Discussion by bruno-grouiez** [earthaccess-dev/earthaccess#470] [Q&A] [answered]
Hi there,
I was thinking of using the `earthaccess.search_data()` method to find out if data is available for OCO-2 for a particular day and given GPS coordinate.
Searching by satellite and day aspect works, but it returns the day granule regardless of the GPS coordinates and the radius of my search circle (same problem with the boundin-box). Here's an example: 

```python
 results = earthaccess.search_data(
        concept_id = "C2716248872-GES_DISC",
        circle=(4.03,49.25,20000),
        temporal=("2022-04-22T09", "2022-04-22T15")
    )
```

Whatever the GPS coordinates or search radius, you'll see that it always returns the granule for the day in question. I thought I'd get 0 granule when the GPS coordinates weren't in a measurement or track zone. 

I have two questions:
1- Does it look for tracks or pixels? I mean, do we search where there is measurement data or where the satellite has passed?
2- What is the operand between the different searches? Is it an AND or an OR and is there a way to choose it?

Finally, does everyone understand why this happens? Is it a bug or is it normal not to have any dependence on the GPS coordinates for a whole day of searching?

Best regards,
Bruno.

**Comment by mfisher87** [comment, marked as answer]
Thanks for your question @bruno-grouiez !

The actual selection of granules is done by [CMR](https://cmr.earthdata.nasa.gov/search/site/docs/search/api.html). It will return any granule which spatially intersects with the passed in bbox, circle, polygon, etc.

In CMR, each granule in this collection is considered to cover the whole earth:

```bash
curl "https://cmr.earthdata.nasa.gov/search/granules.json?collection_concept_id=C2716248872-GES_DISC" | jq .
```

```json
<..snip..>
        "boxes": [
          "-90 -180 90 180"
        ],
<..snip..>
```

A granule covering the whole earth will intersect with any coordinates you provide, so this will always be selected.

If this helps answer your question, please select "mark as answer"!

**Reply by bruno-grouiez** [reply]
Thanks a lot for your reply.
Ok, I now understand that this behavior is actually normal.

In the meantime I've added a step to my routine to test if vertices from oco2 are contained in my perimeter (lat, lon, radius) and it's works, but it's supposed to load the dataset with xarray etc...

Do you see any other way/method to test -without loading dataset- if vertices are in a perimeter for a day?

Regards,
Bruno.

**Reply by mfisher87** [reply]
I don't believe that's possible without loading the dataset, sorry :(

**Reply by bruno-grouiez** [reply]
OK, well thanks anyway
